src/resolvers/ReviewResolver.ts

```diff
@@ -0,0 +1,277 @@
+        return await ReviewCampaign.findOne({
+            where: { id },
+            relations: ["questions", "questions.dropDown", "reviews", "reviews.answers"]
+        });
+    }
+
+    @Query(() => [ReviewCampaign])
+    async getReviewCampaignsForUser(@Arg("eventId") eventId: string, @Arg("ticketType") ticketType?: string) {
+        const campaigns = await ReviewCampaign.find({
+            where: { eventId: { id: eventId } },
+            relations: ["questions", "questions.dropDown"]
+        });
+
+        if (!ticketType) {
+            return campaigns.filter(campaign => campaign.for === "all");
+        }
+
+        return campaigns.filter(campaign =>
+            campaign.for === "all" ||
+            (campaign.for === "ticket_type" && campaign.for_id && campaign.for_id.includes(ticketType))
+        );
+    }
+
+    // Review Campaign Mutations
+    @Mutation(() => CreateReviewCampaignPayload)
+    @UseMiddleware(isAuth)
+    async createReviewCampaign(@Arg("input") input: CreateReviewCampaignInput, @Ctx() { req }: any) {
+        const event = await Events.findOne({ where: { id: input.eventId } });
+        if (!event) {
+            throw new Error("Event not found");
+        }
+
+        const campaign = new ReviewCampaign();
+        campaign.title = input.title;
+        campaign.description = input.description || null;
+        campaign.startDate = input.startDate ? new Date(input.startDate) : new Date();
+        campaign.endDate = input.endDate ? new Date(input.endDate) : new Date();
+        campaign.eventId = event;
+        campaign.for = input.for;
+        campaign.for_id = input.for_id || null;
+
+        await campaign.save();
+
+        // Create questions
+        for (const questionInput of input.questions) {
+            const question = new ReviewQuestion();
+            question.question = questionInput.question;
+            question.questionType = questionInput.questionType;
+            question.rangeMin = questionInput.rangeMin || null;
+            question.rangeMax = questionInput.rangeMax || null;
+            question.reviewCampaign = campaign;
+
+            await question.save();
+
+            // Create dropdown options if needed
+            if (questionInput.dropDown && questionInput.questionType === "dropdown") {
+                for (const dropDownInput of questionInput.dropDown) {
+                    const dropDown = new DropDown();
+                    dropDown.text = dropDownInput.text;
+                    dropDown.value = dropDownInput.value;
+                    dropDown.reviewQuestion = question;
+                    await dropDown.save();
+                }
+            }
+        }
+
+        const result = await ReviewCampaign.findOne({
+            where: { id: campaign.id },
+            relations: ["questions", "questions.dropDown"]
+        });
+
+        return { reviewCampaign: result };
+    }
+
+    @Mutation(() => ReviewCampaign)
+    @UseMiddleware(isAuth)
```

Comment:
Before you can send review_campaign you need to verify user is logged in and authenticated. And your sending them the right review, and not sending them the wrong thing